src/utils.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import print_function
+
+import os
+import io
+import sys
+import imp
+import json
+from os.path import join, exists
+from collections import OrderedDict
+
+ENCODING = 'ISO-8859-1'
+
+
+def is_empty(val):
+    """Check if a variable is an empty string or an empty list."""
+    return val == "" or val == []
+
+
+def clean_input(prompt="", split_char='', ignore_empty=False, dtype=None):
+    """Clean the user-input from the CLI before adding it."""
+    while True:
+        val = input(prompt).strip()
+        # split to list type if split_char specified
+        if split_char != "":
+            val = [v.strip() for v in val.split(split_char) if v.strip() != ""]
+        # do not ignore empty input if not allowed
+        if not ignore_empty and is_empty(val):
+            print("\tError: empty input. Need one or more values.\n")
+            continue
+        # ensure correct input datatype if specified
+        if not is_empty(val) and dtype is not None:
+            try:
+                if not type(eval(val)) == dtype:
+                    print("\tError: input doesn't match required type ", dtype, "\n")
+                    continue
+            except:
+                print("\tError: illegal argument. Input type should be ", dtype, "\n")
+                continue
+        break
+    return val
+
+
+def open_fr(file_name, encoding=ENCODING, encode=True):
```

Comment:
Update, this function with the one in master. In master, we are removing the old python 2 code

Reply:
Also, sorry but doesn't master have the same code?

Reply:
There are currently two versions of open_fr, one in retriever/lib/scripts.py and one in retriever/lib/tools.py. The one in tools should be the one moved away from python 2